service/history/execution/mutable_state_task_generator_test.go

```diff
@@ -83,47 +86,104 @@ func (s *mutableStateTaskGeneratorSuite) TearDownTest() {
 
 func (s *mutableStateTaskGeneratorSuite) TestIsCrossClusterTask() {
 	testCases := []struct {
+		name           string
 		sourceDomainID string
 		targetDomainID string
 		isCrossCluster bool
 		targetCluster  string
+		setupMock      func()
+		err            error
 	}{
 		{
+			name:           "Success case - is not cross cluster task - not cross domain task",
 			sourceDomainID: constants.TestDomainID,
 			targetDomainID: constants.TestDomainID,
 			isCrossCluster: false,
 			targetCluster:  "",
+			setupMock: func() {
+				s.mockMutableState.EXPECT().GetExecutionInfo().Return(&persistence.WorkflowExecutionInfo{
+					DomainID: constants.TestDomainID,
+				}).Times(1)
+			},
 		},
 		{
-			// source domain is passive in the current cluster
+			name:           "Success case - is not cross cluster task - source domain is passive in the current cluster",
 			sourceDomainID: constants.TestRemoteTargetDomainID,
 			targetDomainID: constants.TestDomainID,
 			isCrossCluster: false,
 			targetCluster:  "",
+			setupMock: func() {
+				s.mockMutableState.EXPECT().GetExecutionInfo().Return(&persistence.WorkflowExecutionInfo{
+					DomainID: constants.TestRemoteTargetDomainID,
+				})
+			},
 		},
 		{
+			name:           "Success case - is not cross cluster task - target cluster is the same as source domain active cluster",
```

Comment:
oh no, I have a horrible feeling we're going to collide. That's ok, this is safe to land.